text/3724-theupdateframework.md

```diff
@@ -0,0 +1,341 @@
+
+###### Nightly Role
+
+Nightly releases shall be conducted by a single signing key, trusted for only signing nightly releases of the Rust Compiler. This will allow for nightly releases to remain automated and not require the active participation of any rust members.
+
+###### Rustup Role
+
+This shall be a quorum based role, consisting of all members of the Rustup & Infrastructure team members. We recommend having at least a 3-member threshold. We have decided to have this roles quorum be broader to allow for emergency updates and releases of Rustup; we may want to increase the threshold when these teams have more members.
+
+###### Snapshots & Timestamps Role
+
+These roles shall be a single-member role with a key utilized for automation.
+
+### `rust-lang/tuf-crates`
+
+The actual target for tuf-crates shall be the crates index and not the artifacts themselves. This means that the TUF repository for crates.io is performed on much smaller payloads, which still provides us with cryptographic security due to the fact the index contains SHA-256 hashes of the crate file artifacts.  Given the index already consists of SHA-512 signatures of all files, we are then utilizing TUF to validate the index, which in turn is utilized to validate the actual downloaded artifacts. This allows us to perform validation on index updates and not on final downloads, also reducing the overhead of performing multiple hashing and validation procedures on the larger crate artifact files.
+
+#### Roles
+
+##### Root Role
+
+The root role of the tuf-crates repository shall consist of all members of the crates.io rust team with a threshold of 3. As a special case, updating this role shall also require a resigning by the root role of the tuf-root repository (sign a metadata entry existing within tuf-root). This means any changes to the membership of the crates.io team will also require a signing ceremony via github by the root quorum.
+
+##### Targets Role
+
+The repository shall have a top-level Targets role which is utilized directly for all actions on crates.io. This role shall be a 1-key member role, allowing for the automation of actions by crates.io for publishing of crates. In the future, we propose the delegation of roles for trusted publishing within organizations or crates.
+
+###### Snapshots & Timestamps Roles
+
+These roles shall each be a single-member role with a key utilized for automation.
+
+## TUF Management
+
+We propose the adaptation and implementation of TUF-on-CI (https://github.com/theupdateframework/tuf-on-ci) to manage roots and signing events via GitHub CI. This provides a GitHub-centric workflow for performing signing ceremonies via Pull Requests directly on the TUF repositories in question.
```

Comment:
I think even @jku would say that tuf-on-ci is better suited for the `tuf-root` rather than the `tuf-crates` repo.

For the latter, you probably want to use [Repository Service for TUF (RSTUF)](https://github.com/repository-service-tuf/repository-service-tuf). We are thinking about running RSTUF as a managed service on behalf of OSS package repos like PyPI, and would love to collaborate on testing it for Crates, too.

Cc @kairoaraujo @SantiagoTorres